view/SSViewer.php

```diff
@@ -519,6 +527,64 @@ public function getInjectedValue($property, $params, $cast = true) {
 
 	}
 
+	/**
+	 * Store the current overlay (as it doesn't directly apply to the new scope
+	 * that's being pushed). We want to store the overlay against the next item
+	 * "up" in the stack (hence upIndex), rather than the current item, because
+	 * SSViewer_Scope::obj() has already been called and pushed the new item to
+	 * the stack by this point
+	 * @return SSViewer_Scope
+	 */
+	public function pushScope() {
+		$scope = parent::pushScope();
+
+		$this->itemStack[$this->upIndex][SSViewer_Scope::ITEM_OVERLAY] = $this->overlay;
+		$this->overlay = array();
+
+		return $scope;
+	}
+
+	/**
+	 * Now that we're going to jump up an item in the item stack, we need to
+	 * restore the overlay that was previously stored against the next item "up"
+	 * in the stack from the current one
+	 * @return SSViewer_Scope
+	 */
+	public function popScope() {
+		$this->overlay = $this->itemStack[$this->upIndex][SSViewer_Scope::ITEM_OVERLAY];
+
+		return parent::popScope();
+	}
+
+	/**
+	 * $Up and $Top need to restore the overlay from the parent and top-level
+	 * scope respectively.
+	 */
+	public function obj($name, $arguments = null, $forceReturnedObject = true, $cache = false, $cacheName = null) {
+		$overlayIndex = false;
+
+		switch($name) {
+			case 'Up':
+				if ($this->upIndex === null) {
+					user_error('Up called when we\'re already at the top of the scope', E_USER_ERROR);
```

Comment:
OK - if it's an existing error, then fine... but I think that's a bit harsh when includes could be nested at some unknown level, so stopping access to $Up by way of error instead of just a warning and continuing seems a bit harsh/fragile